packages/backend/src/helpers/flow-templates.ts

```diff
@@ -1,83 +1,80 @@
+import { TEMPLATES } from '@/db/storage'
 import type {
-  AppEventKeyPair,
-  CreateTemplatedFlowInput,
   FlowConfig,
+  TemplateStep,
 } from '@/graphql/__generated__/types.generated'
 import Flow from '@/models/flow'
 import User from '@/models/user'
 
 import logger from './logger'
 
-export const DEFAULT_FLOW_TEMPLATE: CreateTemplatedFlowInput = {
-  flowName: 'Send notifications',
-  trigger: {
-    appKey: 'formsg',
-    eventKey: 'newSubmission',
-  },
-  actions: [
-    {
-      appKey: 'postman',
-      eventKey: 'sendTransactionalEmail',
-    },
-  ],
-  demoVideoId: 'formsg-postman',
-}
-
-async function createFlowFromTemplate(
-  flowName: string,
-  trigger: AppEventKeyPair,
-  actions: AppEventKeyPair[],
+export async function createFlowFromTemplate(
+  templateId: string,
   user: User,
   isAutoCreated: boolean,
-  demoVideoId: string,
 ): Promise<Flow> {
-  const { appKey: triggerAppKey, eventKey: triggerEventKey } = trigger
+  const template = TEMPLATES.find((template) => template.id === templateId)
+  // prevents user from creating any new template
+  if (!template) {
+    throw new Error('Invalid template id input')
+  }
+
+  // check if the template is a demo template first: affects name, config and logging only
+  const { name: flowName, steps, tag } = template
+  const isDemoTemplate = tag === 'demo'
+  const flowConfig: FlowConfig = isDemoTemplate
+    ? {
+        demoConfig: {
+          hasLoadedOnce: false,
+          isAutoCreated,
+          videoId: templateId, // template id is the same as demo video id for backwards compatibility
+        },
+      }
+    : {
+        templateConfig: {
+          templateId,
+        },
+      }
 
   // transaction will insert flow and steps
   return await Flow.transaction(async (trx) => {
-    const flowConfig: FlowConfig = {
-      demoConfig: {
-        hasLoadedOnce: false,
-        isAutoCreated,
-        videoId: demoVideoId,
-      },
-    }
-    // insert flow
+    // insert flow with template id in the demo or template config
     const flow = await user.$relatedQuery('flows', trx).insert({
-      name: `[DEMO] ${flowName}`,
+      name: isDemoTemplate ? `[DEMO] ${flowName}` : flowName,
       config: flowConfig,
     })
 
     // insert trigger step
     await flow.$relatedQuery('steps', trx).insert({
       type: 'trigger',
       position: 1,
-      appKey: triggerAppKey,
-      key: triggerEventKey,
+      appKey: steps[0].appKey,
+      key: steps[0].eventKey,
+      parameters: steps[0].parameters ?? {},
```

Comment:
for sanity check, check that this appKey and key belongs to an actual trigger

Reply:
ok will do so thanks